pkg/commands/workload.go

```diff
@@ -462,13 +495,20 @@ func (opts *WorkloadOptions) PublishLocalSource(ctx context.Context, c *cli.Conf
 
 	// is local source if source is to be created without a source image being specified
 	// or if there is an update of a workload that was initially created to push to LSP
-	isLocal := (opts.SourceImage == "" && currentWorkload == nil) ||
-		(currentWorkload != nil && currentWorkload.IsAnnotationExists(apis.LocalSourceProxyAnnotationName) && opts.SourceImage == "")
+	isLocal := opts.isLocalSource(currentWorkload)
 
 	if isLocal {
 		taggedImage = fmt.Sprintf("%s/%s:%s-%s", source.GetLocalImageRepo(), source.ImageTag, workload.Namespace, workload.Name)
 	} else {
-		taggedImage = workload.Spec.Source.Image
+		if !reflect.DeepEqual(workload.Spec, (cartov1alpha1.WorkloadSpec{})) && workload.Spec.Source != nil {
```

Comment:
question: Why is it necessary to wrap `cartov1alpha1.WorkloadSpec{}` in parenthesis

Reply:
parenthesis removed